plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/local/node/provider/LocalWorkspaceFolderPathProvider.java

```diff
@@ -30,25 +30,64 @@
  */
 @Singleton
 public class LocalWorkspaceFolderPathProvider implements WorkspaceFolderPathProvider {
-    private final String projectsFolderPath;
+
+    @Inject(optional = true)
+    @Named("host.projects.root")
+    /**
+     * this value provide path to projects on local host
+     * if this value will be set all workspace will manage same projects from your host
+     */
+    private String projectsFolder;
+
+    final String workspacesFolder;
 
     @Inject
-    public LocalWorkspaceFolderPathProvider(@Named("host.projects.root") String projectsFolder) throws IOException {
-        Path folder = Paths.get(projectsFolder);
+    public LocalWorkspaceFolderPathProvider(CheHostVfsRootDirProvider workspacesFolderProvider) throws IOException {
+        workspacesFolder = workspacesFolderProvider.get();
+        checkProps(workspacesFolder, projectsFolder);
+    }
+
+    //used for testing
+    protected LocalWorkspaceFolderPathProvider(String workspacesFolder, String projectsFolder) throws IOException{
+        checkProps(workspacesFolder, projectsFolder);
+        this.workspacesFolder = workspacesFolder;
+        this.projectsFolder = projectsFolder;
+    }
+
+    private void checkProps(String workspacesFolder, String projectsFolder) throws IOException {
+        if (workspacesFolder == null && projectsFolder == null) {
+            throw new IOException(
+                    "Can't mount host file system. Check host.workspaces.root or host.projects.root configuration property.");
+        }
+        if (workspacesFolder != null) {
+            ensureExist(workspacesFolder, "host.workspaces.root");
+        }
+        if (projectsFolder != null) {
+            ensureExist(projectsFolder, "host.projects.root");
+        }
+    }
+
+
+    private void ensureExist(String path, String prop) throws IOException {
+        Path folder = Paths.get(path);
         if (Files.notExists(folder)) {
             Files.createDirectory(folder);
         }
         if (!Files.isDirectory(folder)) {
-            throw new IOException("Projects folder " +
-                                  folder.toAbsolutePath() +
-                                  " is invalid. Check vfs.local.fs_root_dir configuration property.");
-
+            throw new IOException(String.format("Projects folder %s is invalid. Check %s configuration property.", path, prop));
```

Comment:
I think that this message is too abstract, it would be really simpler to understand the problem if you include the information that the problem is that given file is not __A DIRECTORY__.